File: output/gpt_4o/avatar/direct_translation/python/java/codeforces_637_A.java

import java.util.*;

public class codeforces_637_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine().trim());
        String[] s = scanner.nextLine().trim().split(" ");
        Map<String, Integer> w = new HashMap<>();
        for (String value : s) {
            w.put(value, w.getOrDefault(value, 0) + 1);
        }
        List<String> sList = Arrays.asList(s);
        Collections.reverse(sList);
        int c = -1;
        String a = "";
        int maxCount = Collections.max(w.values());
        for (String i : w.keySet()) {
            if (w.get(i) == maxCount) {
                int currentIndex = sList.indexOf(i);
                if (currentIndex > c) {
                    a = i;
                    c = currentIndex;
                }
            }
        }
        System.out.println(a);
    }
}

File: output/gpt_4o/avatar/direct_translation/java/python/codeforces_637_A.py

import sys
from collections import defaultdict

def main():
    input = sys.stdin.read
    data = input().split()
    n = int(data[0])
    ara = list(map(int, data[1:n+1]))
    map_count = defaultdict(int)
    
    for v in ara:
        map_count[v] += 1

    max_count = max(map_count.values())

    mm = defaultdict(int)
    for v in ara:
        mm[v] += 1
        if mm[v] == max_count:
            print(v)
            break

if __name__ == "__main__":
    main()

